### PR TITLE
Suicides tracing

### LINKED ISSUE #1635: Trace SUICIDEs

### PR DESCRIPTION
**Pull request by debris**
fixes #1635

### CODE REVIEW

File: ethcore/src/externalities.rs [hunk 1]
```diff
@@ -262,6 +262,8 @@ impl<'a, T, V> Ext for Externalities<'a, T, V> where T: 'a + Tracer, V: 'a + VMT
 			trace!("Suiciding {} -> {} (xfer: {})", address, refund_address, balance);
 			self.state.transfer_balance(&address, refund_address, &balance);
 		}
+
+		self.tracer.trace_suicide(address, balance, refund_address.clone(), self.depth + 1);
```
Comment (debris) [inline ethcore/src/externalities.rs]: I don't like that this line is in externalities, but it seems like the cleanest solution of the problem.

File: ethcore/src/types/trace_types/trace.rs [hunk 2]
```diff
@@ -205,13 +205,57 @@ impl Create {
 	}
 }
 
+/// Suicide action.
+#[derive(Debug, Clone, PartialEq, Binary)]
+pub struct Suicide {
+	/// Suicided address.
+	pub address: Address,
+	/// Suicided contract hier.
```
Comment (rphmeier) [inline ethcore/src/types/trace_types/trace.rs]: typo here?

**Comment by tomusdrw** [comment]
No RPC / serialization tests?

**Comment by gavofyork** [comment]
encode/decode tests for the json is reasonable to do. any actual testing of the RPC itself if rather more involved and should probably go into #863.

**Pull request merged**